test/tests/meshgenerators/output_intermediate_mesh/tests

```diff
@@ -0,0 +1,31 @@
+[Tests]
+  [intermediate_mesh_output]
+  design = 'MeshGenerator.md'
+  issues = '#23486'
+  requirement = 'The system shall be capable of output and check the intermediate mesh object besides the final mesh file'
```

Comment:
These requirements should be as general as possible. The detail here is too specific. While we are outputting only the left block, the _purpose_ of the test is to test this with replicated mesh. How about:

The system shall support the file output of intermediate generated meshes in:
- a serialized form
- a distributed form